File: 71.简化路径.js
/*
 * @lc app=leetcode.cn id=71 lang=javascript
 *
 * [71] 简化路径
 */

// @lc code=start
/**
 * @param {string} path
 * @return {string}
 */
var simplifyPath = function (path) {
    let pathArr = path.split('/');
    let resPath = [];
    for (let i = 0; i < pathArr.length; i++) {
        const name = pathArr[i];
        if (!name) {
            continue;
        } else {
            if (name === '..') {
                resPath.pop();
            } else if (name === '.') {
                continue;
            } else {
                resPath.push(name);
            }
        }
    }

    return '/' + resPath.join('/');
};
// @lc code=end

// console.log(simplifyPath('/home/'));
// console.log(simplifyPath('/../'));
// console.log(simplifyPath('/a/../../b/../c//.//'));
console.log(simplifyPath('/a//b////c/d//././/..'));

File: 71.简化路径.py
#
# @lc app=leetcode.cn id=71 lang=python3
#
# [71] 简化路径
#

# @lc code=start
class Solution:
    def simplifyPath(self, path: str) -> str:
        # 分割
        pathList = path.split('/')
        ans = []
        # 栈
        for name in pathList:
            if not name:
                continue
            else:
                if name == '..':
                    if len(ans) > 0:
                        ans.pop(-1)
                elif name == '.':
                    continue
                else:
                    ans.append(name)
        return '/' + '/'.join(ans)
# @lc code=end

solution = Solution()
ans = solution.simplifyPath('/../')
print(ans)